rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java

```diff
@@ -662,11 +638,296 @@ private void assertInputHasExpectedScriptSig(TransactionInput input, Script rede
+        private static final List<BtcECKey> PROPOSED_FEDERATION_SIGNERS_KEYS =
+            BitcoinTestUtils.getBtcEcKeysFromSeeds(new String[]{"member01", "member02", "member03", "member04", "member05"}, true); // this is needed to have the private keys too
+
+        private static final Keccak256 svpSpendTxCreationHash = RskTestUtils.createHash(1);
+
+        private BtcTransaction svpSpendTx;
+        private List<Sha256Hash> svpSpendTxSigHashes;
+        private List<LogInfo> logs;
+        private BridgeEventLogger bridgeEventLogger;
+
+        @BeforeEach
+        void setUp() {
+            logs = new ArrayList<>();
+            bridgeEventLogger = new BridgeEventLoggerImpl(
+                bridgeMainNetConstants,
+                allActivations,
+                logs
+            );
+
+            bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeMainNetConstants)
+                .withProvider(bridgeStorageProvider)
+                .withActivations(allActivations)
+                .withFederationSupport(federationSupport)
+                .withFeePerKbSupport(feePerKbSupport)
+                .withEventLogger(bridgeEventLogger)
+                .withExecutionBlock(rskExecutionBlock)
+                .build();
+
+            arrangeSvpSpendTransaction();
+            svpSpendTxSigHashes = generateTransactionInputsSigHashes(svpSpendTx);
+        }
+
+        @Test
+        void addSignature_forSvpSpendTx_withWrongKeys_shouldThrowIllegalStateExceptionAndNotAddProposedFederatorSignatures() {
+            // arrange
+            List<BtcECKey> wrongKeys =
+                BitcoinTestUtils.getBtcEcKeysFromSeeds(new String[]{"wrong01", "wrong02", "wrong03", "wrong04", "wrong05"}, true);
+
+            // act & assert
+            for (BtcECKey key : wrongKeys) {
+                List<byte[]> signerEncodedSignatures = generateSignerEncodedSignatures(key, svpSpendTxSigHashes);
+                assertThrows(IllegalStateException.class,
+                    () -> bridgeSupport.addSignature(key, signerEncodedSignatures, svpSpendTxCreationHash));
+            }
+
+            // assert
+            for (BtcECKey key : wrongKeys) {
+                assertFederatorDidNotSignInputs(svpSpendTx.getInputs(), svpSpendTxSigHashes, key);
+            }
+
+            assertAddSignatureWasNotLogged();
+            assertSvpSpendTxWFSWasNotRemoved();
+        }
+
+        @Test
+        void addSignature_forSvpSpendTx_whenProposedFederationDoesNotExist_shouldNotAddProposedFederatorSignatures() throws Exception {
+            // arrange
+            when(federationSupport.getProposedFederation()).thenReturn(Optional.empty());
+
+            // act
+            for (BtcECKey key : PROPOSED_FEDERATION_SIGNERS_KEYS) {
+                List<byte[]> signerEncodedSignatures = generateSignerEncodedSignatures(key, svpSpendTxSigHashes);
+                bridgeSupport.addSignature(key, signerEncodedSignatures, svpSpendTxCreationHash);
```

Comment:
Should not throw an exception when no signature is added?

Reply:
no. In this case `svpIsOngoing` will return false (because theres no proposed federation) so it will try to sign a "normal pegout". But the `rskTxHash` wont be in the `pegoutsWFS` map.